straxen/plugins/peaklet_processing.py

```diff
@@ -168,6 +168,12 @@ def compute(self, records, start, end):
         # Saturation correction using non-saturated channels
         # similar method used in pax
         # see https://github.com/XENON1T/pax/pull/712
+        # Cases when records is not writeable for unclear reason
+        # only see this when loading 1T test data
+        # more details on https://numpy.org/doc/stable/reference/generated/numpy.ndarray.flags.html
+        if not records['data'].flags.writeable:
+            r = records.copy()
```

Comment:
is it the same with `r` instead of `records`? Perhaps a bit clearer if we use `r`?

also don't know what the issue is but I have seen such behavior before too

Reply:
I think yes, here r and records are referencing the same thing, as `r` is here probably just to make the code more succinct

Reply:
Sure, I can change it to r. It just `r=r.copy()` looks unsettling to me, but I did some quick memory profiling, this copying line doesn't even use any new memory. Really have no idea what it's doing internally.